packages/runtime/container-runtime/src/containerRuntime.ts

```diff
@@ -2700,7 +2700,10 @@ export class ContainerRuntime
 				this.updateDocumentDirtyState(false);
```

Comment:
While you're here, I think @dannimad and I should have moved this up to `process` when we [moved the PSM call there](https://github.com/microsoft/FluidFramework/pull/21785). i.e. move this (with the if check) up to `process`, right after the call to `PendingStateManager.processPendingLocalBatch`.

Should we also call it in the non-modern-runtime-message case just to be safe? Kind of annoying, it feels superfluous but would preserve behavior to do so.

(It is slightly odd to me that we update dirty state before actually sending the ack down to the DDS, but it's been this way for like 4 years so whatever.  The point is, this client isn't holding any changes that haven't been saved, so it does make sense in that way.)